src/functional/iterator/embedded.py

```diff
@@ -391,7 +416,7 @@ def execute_shift(
         else:
             raise AssertionError()
         return new_pos
-    elif isinstance(offset_implementation, NeighborTableOffsetProvider):
+    elif isinstance(offset_implementation, NeighborTableOffsetProvider) or isinstance(offset_implementation, NewSparseOffsetProvider):
```

Comment:
The `Connecitivity` is not complete and sufficient for what is used below. Additionally `isinstance` checks with protocols are expensive and I would avoid them in such a performance critical part. I would propose either `isinstance(offset_implementation, (NeighborTableOffsetProvider, NewSparseOffsetProvider))` for now or an extension of `Connectivity` but instead of an `elif` branch just an assert that can be disabled in non-debug mode.